lib/shopify-cli/js_deps.rb

```diff
@@ -74,10 +77,19 @@ def npm(verbose = false)
       deps = %w(dependencies devDependencies).map do |key|
         pkg.fetch(key, []).keys
       end.flatten
+
+      success = true
       CLI::UI::Spinner.spin(ctx.message('core.js_deps.npm_installing_deps', deps.size)) do |spinner|
-        ctx.system(*cmd, chdir: ctx.root)
-        spinner.update_title(ctx.message('core.js_deps.npm_installed_deps', deps.size))
+        success = ctx.system(*cmd, chdir: ctx.root).success?
+
+        if success
+          spinner.update_title(ctx.message('core.js_deps.npm_installed_deps', deps.size))
+        else
+          spinner.update_title(ctx.message('core.js_deps.error.install_error'))
```

Comment:
(Just FYI, in case you didn't know) When the spin block's return value is `CLI::UI::Spinner::TASK_FAILED`, then the spinner title will convert from a `✓` to a `✗` which would avoid that issue you mentioned. 

![image](https://user-images.githubusercontent.com/28009669/84777403-898bce80-afaf-11ea-882e-d813c703c74d.png)

You can see that it also auto debriefs with what went wrong and the output, but you can turn that off with the `auto_debrief: false` option in `.spin`. Though I think its useful info anyway, otherwise the user doesnt really know what went wrong

Reply:
That works!  Great!

For now, I am going to set the `auto_debrief: false` as that more closely aligns with the previous UX. I'm not trying to update how we display errors in the CLI just make sure we display error cases when they actually occur.

New output for npm failure:
![Screen Shot 2020-06-16 at 9 13 40 AM](https://user-images.githubusercontent.com/42751082/84779050-dffa0c80-afb1-11ea-82f8-8d6d987acc93.png)